app_dart/lib/src/request_handlers/test_ownership.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:cocoon_service/src/request_handlers/flaky_handler_utils.dart';
+
+abstract class TestOwner {
+  factory TestOwner(BuilderType builderType) {
+    switch (builderType) {
+      case BuilderType.devicelab:
+        return DeviceLabTestOwner();
+      case BuilderType.firebaselab:
+        return FirebaseLabTestOwner();
+      case BuilderType.frameworkHostOnly:
+        return FrameworkHostOnlyTestOwner();
+      case BuilderType.shard:
+        return ShardTestOwner();
+      default:
+        return UnknownTestOwner();
+    }
+  }
+
+  TestOwnership getTestOwnership(String targetName, String testOwnersContent);
+}
+
+Team teamFromString(String teamString) {
+  switch (teamString) {
+    case 'flutter/framework':
+      return Team.framework;
+    case 'flutter/engine':
+      return Team.engine;
+    case 'flutter/tool':
+      return Team.tool;
+    case 'flutter/web':
+      return Team.web;
+  }
+  return Team.unknown;
+}
+
+String getTestNameFromTargetName(String targetName) {
+  // The builder names is in the format '<platform> <test name>'.
+  final List<String> words = targetName.split(' ');
+  return words.length < 2 ? words[0] : words[1];
+}
+
+class DeviceLabTestOwner implements TestOwner {
+  DeviceLabTestOwner();
+
+  @override
+  TestOwnership getTestOwnership(String targetName, String testOwnersContent) {
+    String? owner;
+    Team? team;
+    final String testName = getTestNameFromTargetName(targetName);
```

Comment:
We should use test name directly instead of parsing the test name based on target name.

Counter example:
target name: `Mac_ios spell_check_test`
task name: `spell_check_test_ios`

Reply:
It is for the sake of the TESTOWNER validation since I am passing in just the task name from the validate test ownership file. However if this is to be applied everywhere then this will take a significant amount of time as there are multiple places we are calling this from where the task name might no be available. I will have to look again.

Reply:
I received clarification in a gvc.